cawemo-to-web-modeler-migration/migrate-cawemo-to-web-modeler.sh

```diff
@@ -0,0 +1,210 @@
+#!/bin/sh 
+
+# ℹ️ Requires Bash 4 or newer, or zsh
+# Recommended to run it on a test environment first
+
+# See these docs to obtain your Cawemo API credentials: https://docs.camunda.org/cawemo/1.9/reference/rest-api/overview/authentication/
+CAWEMO_USER_ID=ENTER_HERE
+CAWEMO_API_KEY=ENTER_HERE
+
+# See these docs to obtain your Web Modeler client credentials: https://docs.camunda.io/docs/next/apis-tools/web-modeler-api/#authentication
+MODELER_CLIENT_ID=ENTER_HERE
+MODELER_CLIENT_SECRET=ENTER_HERE
+
+# Color definitions for terminal output formatting
+GREEN='\033[0;32m'
+NC='\033[0m' # No Color
+B=$(tput bold)
+N=$(tput sgr0)
+BGREEN=$GREEN$B
+NNC=$NC$N
+
+echo "------------------------------------------------------------------------"
+echo "${BGREEN}MIGRATION FROM CAWEMO TO WEB MODELER${NNC}"
+echo "Files are now migrated file-by-file..."
+echo "------------------------------------------------------------------------"
+
+echo ""
+
+echo "[$(date +%s)] Migration started" >> migration.log
+
+HTTP_STATUS=$(curl -s -o .curl_tmp -w "%{http_code}" -u $CAWEMO_USER_ID:$CAWEMO_API_KEY "https://cawemo.com/api/v1/projects")
+
+PROJECTS=$(cat .curl_tmp)
+
+if [ "$HTTP_STATUS" != "200" ]
+then
+    echo "⚠ GET https://cawemo.com/api/v1/projects failed with status code $HTTP_STATUS"
+    echo "[$(date +%s)] ⚠ GET https://cawemo.com/api/v1/projects failed with status code $HTTP_STATUS" >> migration.log
+    
+    echo "Aborted"
+    exit 1
+fi
+
+TOKEN=$(curl -s --header "Content-Type: application/json" --request POST --data "{\"grant_type\":\"client_credentials\", \"audience\":\"api.cloud.camunda.io\", \"client_id\":\"$MODELER_CLIENT_ID\", \"client_secret\":\"$MODELER_CLIENT_SECRET\"}" https://login.cloud.camunda.io/oauth/token | jq -r .access_token)
+
+for row in $(echo $PROJECTS | jq -r '.[] | @base64'); do
+    _jq() {
+     echo ${row} | base64 --decode | jq -r ${1}
+    }
+
+    OLD_PROJECT_ID=$(_jq '.id')
+    PROJECT_NAME=$(_jq '.name')
+
+    echo "Migrating project ${GREEN}$PROJECT_NAME${NC} (ID: $OLD_PROJECT_ID)..."
+
+    HTTP_STATUS=$(curl -s -o .curl_tmp -w "%{http_code}" -X POST "https://modeler.cloud.camunda.io/api/beta/projects" \
+        -H "Content-Type: application/json" \
+        -H "Authorization: Bearer $TOKEN" \
+        -d "$(_jq '.')")
+
+    NEW_PROJECT=$(cat .curl_tmp)
+
+    if [ "$HTTP_STATUS" != "200" ]
+    then
+        echo "⚠ POST https://modeler.cloud.camunda.io/api/beta/projects failed with status code $HTTP_STATUS"
+        echo "[$(date +%s)] ⚠ POST https://modeler.cloud.camunda.io/api/beta/projects failed with status code $HTTP_STATUS" >> migration.log
+
+        echo "Aborted"
+        exit 1
+    fi
+
+    echo "[$(date +%s)] Migrated project $PROJECT_NAME (ID: $OLD_PROJECT_ID)" >> migration.log
+
+    NEW_PROJECT_ID=$(jq -r .id <<< "$NEW_PROJECT")
+
+    # maps folder IDs to original IDs to not create them twice + create file in right folder
+    declare -A folderIds
```

Comment:
Does not seem to work in Zsh on macOS:
```
./migrate-cawemo-to-web-modeler.sh: line 77: declare: -A: invalid option
declare: usage: declare [-afFirtx] [-p] [name[=value] ...]
```

Reply:
Only happens when I run the script directly with `./migrate-cawemo-to-web-modeler.sh`, not with `zsh migrate-cawemo-to-web-modeler.sh` 🤔.

Reply:
I guess `declare` does not exist in `sh` the shell specified by the shebang at the top of the script.

To align behaviours the first line should be changed to `#!/usr/bin/env zsh`

Reply:
Works now when using `#!/bin/zsh`

@jfriedenstab please verify